### PR TITLE
Update package versions, remove dead code

### LINKED ISSUE #44: Using this web component breaks the usage of web3.utils
I am trying to use this web component in my DApp, and after including it, web3.utils is undefined.
image

I understand that you are loading web3, it would be great if you could encapsulate it or have the application decide to bring its own (e.g. by setting a parameter to the component data-own-web3-provider)

I am using web3 from this provider (at the time of writing)
https://cdn.jsdelivr.net/npm/web3@latest/dist/web3.min.js

And connect it in this way:
window.web3 = new Web3(window.ethereum);

### PR DESCRIPTION
**Pull request by fafrd**
Three things in this PR:
1. Update a _lot_ of packages. Some things are still not quite up to date, like webpack, which had breaking changes between version 4 and 5... and tslint, which is deprecated, it should be replaced with eslint but this was a bit involved (there seem to be some [major issues](https://github.com/typescript-eslint/typescript-eslint/issues/1824) with their indent rule??)
2. Did a bit of refactoring to kill off dead code. This package used to have functionality to allow users to buy an NFT directly from the embed, but this was partly removed in c09a685; in this PR i've fully removed the remaining functionality.
3. Fix the flip card functionality- [example of how it's broken](https://opensea.io/blog/announcements/announcing-embeddable-nfts/). We use an obscure css property called `backface-visibility`, I think this used to not be properly supported in browsers but now it is. Fixed by passing down the 'is-flipped' state to the child front/back components.

Resolves #44, #29, #31, #22, #25, #28, #30, #35, #41, #38

### CODE REVIEW

**Review by c-h- — APPROVED** [review]
Are you able to share more specifics about how you tested the dependency upgrades and the backface update, and how you ensured the dead code is dead?

**Review by michael-cohen-eth — APPROVED** [review]

File: src/nft-card-front.ts [hunk 1]
```diff
@@ -316,58 +313,9 @@ export class NftCardFrontTemplate extends LitElement {
   private getButtonTemplate() {
     return html`
       <button @click="${(e: any) => this.eventHandler(e, 'view')}">
-        ${BTN_TEXT[ButtonType.Buy]}
+        buy this item ❯
```
Comment (sunwrobert) [inline src/nft-card-front.ts:316]: don't we normally uppercase the first letter?

**Review by sunwrobert — APPROVED** [review]

File: src/nft-card-front.ts (hunk 1 above)
Comment (fafrd) [reply src/nft-card-front.ts:316]: this element has text-transform: uppercase, so it's irrelevant here

**Comment by fafrd** [comment]
> Are you able to share more specifics about how you tested the dependency upgrades and the backface update, and how you ensured the dead code is dead?

Essentially I just tested this out in a few browsers (chrome, firefox and safari) and ensured everything worked manually. It's a basic enough project that I feel pretty confident in this.

**Pull request merged**